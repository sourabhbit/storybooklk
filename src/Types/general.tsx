export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
