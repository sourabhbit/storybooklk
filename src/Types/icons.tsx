export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
