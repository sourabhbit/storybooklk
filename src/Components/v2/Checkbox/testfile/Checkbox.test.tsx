import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, TypographyENUM } from 'Types/general'
import CheckBox from '../index'

const props = {
	label: 'black',
	checked: false,
	size: ComponentSizeENUM.medium,
	font: TypographyENUM.serif,
	onChange: jest.fn(),
}

describe('<CheckBox />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<CheckBox {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('checkbox-test')
		expect(container).toBeInTheDocument()
	})

	// it('Should call function onClick', () => {
	// 	const mainDrop = screen.getByTestId('checkbox-test')
	// 	userEvent.click(mainDrop)
	// })

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
