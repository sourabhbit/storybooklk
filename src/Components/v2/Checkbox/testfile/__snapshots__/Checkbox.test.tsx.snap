// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL dYfRsX"
  >
    <input
      class="sc-gsDKAQ fJRCyr"
      data-testid="checkbox-test"
      type="checkbox"
    />
    <span
      class="sc-dkPtRN kwqbju"
    >
      black
    </span>
  </div>
</DocumentFragment>
`;
